notebooks/3.jump-analysis/3.jump_analysis.py

```diff
@@ -0,0 +1,453 @@
+shuffled_y_proba_df["shuffled_model"] = True
+
+shuffled_gt_y_proba_df = pd.DataFrame(shuffled_gt_y_proba)
+shuffled_gt_y_proba_df["pred_injury"] = shuffled_gt_y_pred.flatten()
+shuffled_gt_y_proba_df["datatype"] = "JUMP Overlap"
+shuffled_gt_y_proba_df["shuffled_model"] = True
+
+# concatenate all prediction
+# update the predicted label columns to injury name
+all_proba_scores = pd.concat(
+    [y_proba_df, gt_y_proba_df, shuffled_y_proba_df, shuffled_gt_y_proba_df]
+)
+all_proba_scores.columns = [
+    injury_decoder[str(col_name)] for col_name in all_proba_scores.columns[0:15]
+] + col_to_sel
+all_proba_scores["pred_injury"] = all_proba_scores["pred_injury"].apply(
+    lambda injury_code: injury_decoder[str(injury_code)]
+)
+
+# next only select cytoskeletal probability scores
+cytoskeletal_proba_scores = all_proba_scores[col_to_sel + ["Cytoskeletal"]]
+cytoskeletal_proba_scores.rename(columns={"Cytoskeletal": "Cytoskeletal Proba"})
+
+
+# In[14]:
+
+
+# Saving only cyr
+cytoskeletal_proba_scores.to_csv(
+    jump_analysis_dir / "cytoskeletal_proba_scores.csv.gz",
+    compression="gzip",
+    index=False,
+)
+
+
+# In[15]:
+
+
+# making all probabilities tidy long
+all_injury_proba = all_proba_scores[col_to_sel + injury_classes].melt(
+    id_vars=["pred_injury", "datatype", "shuffled"],
+    var_name="injury_compared_to",
+    value_name="proba",
+)
+
+# save file
+all_injury_proba.to_csv(
+    jump_analysis_dir / "all_injury_proba.csv.gz", index=False, compression="gzip"
+)
+all_injury_proba.head()
+
+
+# ## Generating Confusion Matrix
+
+# In[16]:
+
+
+shared_treat_meta, shared_treat_feats = split_meta_and_features(shared_treat_jump_df)
+shared_X = shared_treat_jump_df[shared_treat_feats]
+shared_y = shared_treat_jump_df["injury_code"]
+
+
+# In[17]:
+
+
+jump_overlap_cm = generate_confusion_matrix_tl(
+    model, shared_X, shared_y, shuffled=False, dataset_type="JUMP Overlap"
+).fillna(0)
+shuffled_jump_overlap_cm = generate_confusion_matrix_tl(
+    shuffled_model, shared_X, shared_y, shuffled=True, dataset_type="JUMP Overlap"
+).fillna(0)
+
+
+# In[18]:
+
+
+# save confusion matrix
+pd.concat([jump_overlap_cm, shuffled_jump_overlap_cm]).to_csv(
+    modeling_dir / "jump_overlap_confusion_matrix.csv.gz",
+    compression="gzip",
+    index=False,
+)
+
+
+# ## Creating supplemental Table
+
+# In[19]:
+
+
+shared_jump_meta, shared_jump_feats = split_meta_and_features(shared_jump_df)
+
+
+# In[20]:
+
+
+jump_meta = shared_jump_df[["Metadata_Plate", "Metadata_Well", "Metadata_pert_iname"]]
+jump_meta["pred_injury"] = [
+    injury_decoder[str(injury_code)] for injury_code in y_pred.tolist()
+]
+jump_meta["probability"] = y_proba.max(axis=1).tolist()
+
+
+# In[21]:
+
+
+# save supplemental figure
+jump_meta.to_csv(
+    jump_analysis_dir / "stable2_predicted_jump_injury_table.csv.gz",
+    compression="gzip",
+    index=False,
+)
+jump_meta
+
+
+# In[22]:
+
+
+jump_meta.loc[jump_meta["Metadata_Plate"] == "BR00117006"][
```

Comment:
Without sounding rude, why is this here after exporting the data? If it is there for a reason consider documenting it

Reply:
Not at all sounding! However....Ehhh... This was not supposed be here. I thought I deleted this cell... I might have hit the undo button too many time and brought the cell from the dead. 😨 

Thanks for the catch! Fixed! ✅